NetKet/GroundState/ground_state.hpp

```diff
@@ -12,102 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef NETKET_GROUND_STATE_CC
-#define NETKET_GROUND_STATE_CC
+#ifndef NETKET_GROUND_STATE_HPP
+#define NETKET_GROUND_STATE_HPP
 
 #include <memory>
+#include <string>
+#include <vector>
 
 #include "Hamiltonian/MatrixWrapper/matrix_wrapper.hpp"
 #include "Observable/observable.hpp"
 #include "Optimizer/optimizer.hpp"
 
 #include "imaginary_time.hpp"
 #include "variational_montecarlo.hpp"
+#include "exact_diagonalization.hpp"
 
 namespace netket {
 
-class GroundState {
- public:
-  explicit GroundState(const json &pars) {
-    std::string method_name;
-
-    if (FieldExists(pars, "GroundState")) {
-      method_name = FieldVal(pars["GroundState"], "Method", "GroundState");
-    } else if (FieldExists(pars, "Learning")) {
-      method_name = FieldVal(pars["Learning"], "Method", "Learning");
-      // DEPRECATED (to remove for v2.0.0)
-      WarningMessage()
+  class GroundState {
```

Comment:
The indentation is not consistent with the style guide. I suggest you run `clang-format` as describe in https://github.com/netket/netket/blob/master/CONTRIBUTING.md#c-coding-style on the files changed in this PR.